src/test/librbd/test_mock_fixture.h

```diff
@@ -25,7 +25,11 @@ ACTION_P(CopyInBufferlist, str) {
 }
 
 ACTION_P2(CompleteContext, r, wq) {
+#if defined(__clang__)
+  ContextWQ *context_wq = static_cast<ContextWQ *>(wq);
```

Comment:
@wjwithagen seems on FreeBSD the C land is still using the FreeBSD's own libc. see 

- stdef.h from llvm/libc++: https://github.com/llvm-mirror/clang/blob/master/lib/Headers/stddef.h and 
- stddef.h from FreeBSD. https://github.com/freebsd/freebsd/blob/386ddae58459341ec567604707805814a2128a57/sys/sys/stddef.h
  - and included _null.h: https://github.com/freebsd/freebsd/blob/386ddae58459341ec567604707805814a2128a57/sys/sys/_null.h

in other words, FreeBSD folks think that `NULL == nullptr` with C++11 or up. per N3337.

§ 18.2
> The macro NULL is an implementation-defined C++ null pointer constant in this International Standard (4.10).194

§ 4.10
> 4.10 Pointer conversions [conv.ptr]
> A null pointer constant is an integral constant expression (5.19) prvalue of integer type that evaluates to zero **or** a prvalue of type std::nullptr_t. A null pointer constant can be converted to a pointer type; the result is the null pointer value of that type and is distinguishable from every other value of object pointer or function pointer type. Such a conversion is called a null pointer conversion. Two null pointer values of the same type shall compare equal. The conversion of a null pointer constant to a pointer to cv-qualified type is a single conversion, and not the sequence of a pointer conversion followed by a qualification conversion (4.4).
> A null pointer constant of integral type can be converted to a prvalue of type std::nullptr_t. [ Note: The resulting prvalue is not a null pointer value. — end note ]

so i guess that FreeBSD's implementation of NULL makes sense. as it choose to use "a prvalue of type std::nullptr_t" instead of "an integral constant expression (5.19) prvalue of integer type that evaluates to zero". and the former is also acceptable per the C++11 standard.

@dillaman does this make sense?